src/datachain/lib/listing.py

```diff
@@ -99,8 +104,13 @@ def _isfile(client: "Client", path: str) -> bool:
             return False
 
         return info["type"] == "file"
-    except:  # noqa: E722
+    except FileNotFoundError:
         return False
+    except REMOTE_ERRORS as e:
+        raise ClientError(
+            message=str(e),
+            error_code=getattr(e, "code", None),
+        ) from e
```

Comment:
Do we want to catch other exceptions here? 🤔

Reply:
I think that should be the burden of outer call or calling function to capture exception that are totally unexpected.